util/metric/metric.go

```diff
@@ -205,15 +232,29 @@ func (c *Counter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Counter.Count())
 }
 
+// Desc implements prometheus.Metric.
+func (c *Counter) Desc() *prometheus.Desc {
+	return c.desc
+}
+
+// Write implements prometheus.Metric.
+func (c *Counter) Write(met *prometheus_model.Metric) error {
+	met.Gauge = &prometheus_model.Gauge{Value: proto.Float64(float64(c.Count()))}
```

Comment:
You are operating on the protobuf level here. Nothing wrong with that, of course. It's rather efficient, even. It's relatively easy because you don't add any labels here.

The more high-level way would be

```prometheus.MustNewConstMetric(c.desc, prometheus.GaugeValue, float64(c.Count()).Write(met)```

which has the advantage that you can easily add labels later:

```prometheus.MustNewConstMetric(c.desc, prometheus.GaugeValue, float64(c.Count(), "foo", "bar").Write(met)```

Different question: Are your counters not counters in the Prometheus sense? If yes, you can use the Prometheus Counter type here.